bin/oref0-pump-loop.sh

```diff
@@ -152,16 +152,28 @@ function smb_reservoir_before {
     # Refresh reservoir.json and pumphistory.json
     try_fail refresh_pumphistory_and_meal
     try_fail cp monitor/reservoir.json monitor/lastreservoir.json
+    CURRENT_DTG=$(date +%s)
     try_fail timerun openaps report invoke monitor/clock.json monitor/clock-zoned.json 2>&3 >&4 | tail -1
     echo -n "Checking pump clock: "
     (cat monitor/clock-zoned.json; echo) | tr -d '\n'
     echo -n " is within 90s of current time: " && date
-    if (( $(bc <<< "$(date +%s -d $(cat monitor/clock-zoned.json | sed 's/"//g')) - $(date +%s)") < -55 )) || (( $(bc <<< "$(date +%s -d $(cat monitor/clock-zoned.json | sed 's/"//g')) - $(date +%s)") > 55 )); then
-        echo Pump clock is more than 55s off: attempting to reset it
+    let DTG_DIFFERENCE=$CURRENT_DTG-$(date +%s -d $(cat monitor/clock-zoned.json | sed 's/"//g'))
+    DTG_DIFFERENCE=${DTG_DIFFERENCE/#-/}
+    if [ "$DTG_DIFFERENCE" -gt "90" ]
```

Comment:
This has to be less than 90 to avoid a boundary condition. I chose 55.

Reply:
Specifically, we ran into a situation where when it was set to 90 seconds it would not reset the clock but would also feel the following check.

Reply:
So instead I had it try to fix the clock any time it is 55s off but keep looping if that failed, up to 90s.

Reply:
I can't be sure, but I believe that the difference in check-design might fix that...

Because you were obtaining pump-time at the beginning (openaps call) then calling openaps again to zone it, then getting the 'current' time as part of a series of sub-shells in the calculate routine, it was possible for enough time to pass to get a false-fail...

I would put forward that the issue I was experincing with the neutered PiZero was simply that at 500mhz, it was running so slowly that I slipped outside the 55s-vs-90s window on a continuous basis. 

The changes above should eliminate that edge-case unless the system is so incredibly bogged down that it takes 90 seconds between 'date +%s' populating CURRENT_DTG, & openaps-invoke & monitor/clock.json being written to disk.  

Which is logically harder to have happen, than it is for 90s to pass between clock.json being written, and the process of writing out clock-zoned & executing the bc statement with it's sub-shells.

Reply:
Ok, that's good, but it gets us back to the original reason for the 55s pump clock set: the time of the last bolus in pumphistory is what's used to determine if it's been long enough to SMB again.  If we relax the enforcement to allow the pump clock 90s off of the rig time (instead of requiring that the offset minus the time it takes to do the check doesn't exceed 90s), then we can be in a situation where we're SMB'ing every 2 minutes, which is probably too fast.  So it's better to try to reset the pump clock if it has drifted more than a minute, even if we still allow the loop to proceed up to a longer limit.

Reply:
Got it. So check 55s on the first one, 90 on the 2nd is what you would prefer.